crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -89,7 +89,9 @@ pub(crate) enum DefinitionNodeRef<'a> {
     AnnotatedAssignment(&'a ast::StmtAnnAssign),
     AugmentedAssignment(&'a ast::StmtAugAssign),
     Comprehension(ComprehensionDefinitionNodeRef<'a>),
-    Parameter(ast::AnyParameterRef<'a>),
+    VariadicParameter(&'a ast::Parameter),
+    VariadicKeywordParameter(&'a ast::Parameter),
+    ParameterWithDefault(&'a ast::ParameterWithDefault),
```

Comment:
nit: for consistency, I'd probably call these variants

```suggestion
    VariadicPositionalParameter(&'a ast::Parameter),
    VariadicKeywordParameter(&'a ast::Parameter),
    NonVariadicParameter(&'a ast::ParameterWithDefault),
```

Reply:
It's not clear to me that this is an improvement. I find it weird to call the common-case parameter `NonVariadicParameter`: defining it by what it isn't, even though it's the much more common case, and variadic parameters are the unusual case.

And similarly, I think the default assumption for "variadic" is "a variable-length list of items" (not a keyword mapping), so again I find it just as clear to just call that `VariadicParameter` and emphasize the unusual case of the variadic keyword parameter.

Reply:
> I find it weird to call the common-case parameter `NonVariadicParameter`: defining it by what it isn't, even though it's the much more common case, and variadic parameters are the unusual case.

I agree `NonVariadicParameter` isn't a great name, but it still feels better to me than `ParameterWithDefault`, considering that most `ParameterWithDefault`s don't have defaults.

> And similarly, I think the default assumption for "variadic" is "a variable-length list of items" (not a keyword mapping)

To me "variadic" could well refer to either! I don't agree there's a default assumption there

Reply:
Anyway, these are nitpicks — I obviously don't feel strongly!

Reply:
I can switch to `VariadicPositionalParameter`.

How do you feel about simply `Parameter` for the normal-parameter case?